sermant-plugins/sermant-mq-grayscale/mq-grayscale-rocketmq-plugin/src/main/java/io/sermant/mq/grayscale/rocketmq/service/RocketMqConsumerGroupAutoCheck.java

```diff
@@ -0,0 +1,227 @@
+import io.sermant.mq.grayscale.config.GrayTagItem;
+import io.sermant.mq.grayscale.rocketmq.config.RocketMqConsumerClientConfig;
+import io.sermant.mq.grayscale.rocketmq.utils.RocketMqGrayscaleConfigUtils;
+import io.sermant.mq.grayscale.rocketmq.utils.RocketMqSubscriptionDataUtils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.rocketmq.client.exception.MQBrokerException;
+import org.apache.rocketmq.client.exception.MQClientException;
+import org.apache.rocketmq.client.impl.MQClientAPIImpl;
+import org.apache.rocketmq.client.impl.factory.MQClientInstance;
+import org.apache.rocketmq.common.protocol.body.GroupList;
+import org.apache.rocketmq.common.protocol.route.BrokerData;
+import org.apache.rocketmq.common.protocol.route.TopicRouteData;
+import org.apache.rocketmq.remoting.exception.RemotingConnectException;
+import org.apache.rocketmq.remoting.exception.RemotingSendRequestException;
+import org.apache.rocketmq.remoting.exception.RemotingTimeoutException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * consumer group auto check service
+ *
+ * @author chengyouling
+ * @since 2024-05-27
+ **/
+public class RocketMqConsumerGroupAutoCheck {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    private static final ScheduledExecutorService EXECUTOR_SERVICE = Executors.newSingleThreadScheduledExecutor();
+
+    /**
+     * gray consumer tags at last time
+     * key: namesrvAddr@topic@consumerGroup
+     * value: consumer gray tags
+     */
+    private static final Map<String, Set<String>> LAST_TOPIC_GROUP_GRAY_TAG = new HashMap<>();
+
+    /**
+     * client configs for query consumer group
+     * key: namesrvAddr@topic@consumerGroup
+     * value: client config
+     */
+    private static final Map<String, RocketMqConsumerClientConfig> CONSUMER_CLIENT_CONFIG_MAP = new HashMap<>();
+
+    private static final AtomicBoolean START_AUTO_CHECK = new AtomicBoolean(false);
+
+    private static final long INITIAL_DELAY = 10L;
+
+    private static final long ROCKET_MQ_READ_TIMEOUT = 5000L;
+
+    private RocketMqConsumerGroupAutoCheck() {
+    }
+
+    /**
+     * set current client MQClientInstance info
+     *
+     * @param topic topic
+     * @param consumerGroup consumerGroup
+     * @param mqClientInstance mqClientInstance
+     */
+    public static void setMqClientInstance(String topic, String consumerGroup, MQClientInstance mqClientInstance) {
+        String addrTopicGroupKey = RocketMqSubscriptionDataUtils.buildAddrTopicGroupKey(topic, consumerGroup,
```

Comment:
The responsibilities of the method are not single enough.

Reply:
fixed